frontend/src/components/Modal.js

```diff
@@ -11,78 +11,67 @@ import {
   Label,
 } from "reactstrap";
 
-export default class CustomModal extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      activeItem: this.props.activeItem,
-    };
-  }
+export const CustomModal = ({inputItem, toggle, onSave}) => {
+  const [activeItem, setActiveItem] = useState(inputItem)
 
-  handleChange = (e) => {
+  const handleChange = (e) => {
     let { name, value } = e.target;
 
     if (e.target.type === "checkbox") {
       value = e.target.checked;
     }
 
-    const activeItem = { ...this.state.activeItem, [name]: value };
-
-    this.setState({ activeItem });
+    setActiveItem({ ...activeItem, [name]: value });
   };
 
-  render() {
-    const { toggle, onSave } = this.props;
-
-    return (
-      <Modal isOpen={true} toggle={toggle}>
-        <ModalHeader toggle={toggle}>Todo Item</ModalHeader>
-        <ModalBody>
-          <Form>
-            <FormGroup>
-              <Label for="todo-title">Title</Label>
-              <Input
-                type="text"
-                id="todo-title"
-                name="title"
-                value={this.state.activeItem.title}
-                onChange={this.handleChange}
-                placeholder="Enter Todo Title"
-              />
-            </FormGroup>
-            <FormGroup>
-              <Label for="todo-description">Description</Label>
+  return (
+    <Modal isOpen={true} toggle={toggle}>
+      <ModalHeader toggle={toggle}>Todo Item</ModalHeader>
+      <ModalBody>
+        <Form>
+          <FormGroup>
```

Comment:
We can also create a component called IDK `FormGroup` and accepts the label string and the necessary props for the input component. This way re-use it in the three input fields that we have.

Reply:
So I've been thinking about this and wonder if it might be an over-optimization at this point. Since only 2 of the inputs are similar enough to make this really work, is there a large enough benefit to warrant the new component?